### PR TITLE
feat: update nx to v19.0.6

### PR DESCRIPTION
**Pull request by arvidboivie**
Hey, figured I'd give a shot at updating this for Nx 19. 

This generates a couple of dependency warnings:

```
 WARN  Issues with peer dependencies found
.
├─┬ @jscutlery/semver 5.2.0
│ └── ✕ unmet peer @nx/devkit@^18.0.0: found 19.0.4
└─┬ ngx-deploy-npm 7.1.0
  └── ✕ unmet peer @nx/devkit@"^16.0.0 || ^17.0.0 || ^18.0.0": found 19.0.4
```

`@jscutlery/semver` seems to have a an update on the way: https://github.com/jscutlery/semver/issues/814
Not sure about `ngx-deploy-npm` though, it support 18 but is still on 17 itself.

### CODE REVIEW

**Review by robinpellegrims — CHANGES_REQUESTED** [review]
Looks good to me, thanks for your contribution!
There were no automatic migrations to run?
Could you also mention BREAKING CHANGE in the commit message to trigger a major version update?

File: pnpm-lock.yaml
Comment (robinpellegrims) [inline pnpm-lock.yaml:1]: seems like this file is incorrectly formatted, could you fix it?

**Comment by robinpellegrims** [comment]
Could you also fix the following lint error? The nx-remotecache-custom version in the buildable libs package.json needs to match the new version in the root package.json.
`/libs/nx-remotecache-s3/package.json
  41:5  error  The version specifier does not contain the installed version of "nx-remotecache-custom" package: 19.0.0  @nx/dependency-checks`

**Comment by arvidboivie** [comment]
No automatic migrations, the only breaking change in NX was a change in how they bundle env variables: https://nx.dev/blog/2024-05-08-nx-19-release#breaking-change-updating-bundled-environment-variables-to

Will adress your comments during the day!

File: libs/nx-remotecache-s3/package.json [hunk 1]
```diff
@@ -30,15 +30,15 @@
   },
   "homepage": "https://github.com/robinpellegrims/pellegrims/tree/master/libs/nx-remotecache-s3",
   "peerDependencies": {
-    "nx": ">=17.0.0"
+    "nx": ">=19.0.0"
```
Comment (arvidboivie) [inline libs/nx-remotecache-s3/package.json:33]: This was lagging behind a version, I think it would make sense it goes up to 19 as well?

**Comment by arvidboivie** [comment]
Not sure what you want to with `ngx-deploy-npm` and `@jscutlery/semver`, do we wait for them to support 19?

**Comment by robinpellegrims** [comment]
> Not sure what you want to with `ngx-deploy-npm` and `@jscutlery/semver`, do we wait for them to support 19?

Ah yes, I forgot to check if it will be a problem. I'll try and do this later today.

**Comment by robinpellegrims** [comment]
> > Not sure what you want to with `ngx-deploy-npm` and `@jscutlery/semver`, do we wait for them to support 19?
> 
> Ah yes, I forgot to check if it will be a problem. I'll try and do this later today.

Looks good, both dependencies seem to work with the newest version (apart from the warning) so we should be able to upgrade them later.

Thanks again!

**Pull request merged**